src/frontend/utils/telemetry/InstrumentationMiddleware.ts

```diff
@@ -1,39 +1,51 @@
-import { NextApiHandler } from 'next';
+import {NextApiHandler} from 'next';
 import Tracer from './BackendTracer';
-import { context, propagation, SpanKind, SpanStatusCode, Exception } from '@opentelemetry/api';
-import { SemanticAttributes } from '@opentelemetry/semantic-conventions';
+import {context, Exception, propagation, SpanKind, SpanStatusCode, trace} from '@opentelemetry/api';
+import {SemanticAttributes} from '@opentelemetry/semantic-conventions';
+import {Span} from '@opentelemetry/sdk-trace-base';
 
 const InstrumentationMiddleware = (handler: NextApiHandler): NextApiHandler => {
-  const wrapper: NextApiHandler = async (request, response) => {
-    const { headers, method, url = '', httpVersion } = request;
-    const [target] = url.split('?');
+    return async (request, response) => {
+        const {headers, method, url = '', httpVersion} = request;
+        const [target] = url.split('?');
 
-    const parentContext = propagation.extract(context.active(), headers);
-    const span = await Tracer.createSpanFromContext(`API HTTP ${method}`, parentContext, { kind: SpanKind.SERVER });
+        let span;
+        const baggage = propagation.getBaggage(context.active());
```

Comment:
nit: I think we can move most of this span creation logic inside of the BackendTracer functionality, we an just keep the span assignment

Reply:
If we moved this to the BackendTracer, we would still need to do all the attributes in the Middleware.